### ISSUE TITLE
Unsupervised Clustering forest

### ISSUE THREAD

**Issue by lizequn** [dmlc/xgboost#567]
Hi,
I am new to Xgboost and I found the training data must contain labels, So I want to figure out if there's a easy way to implement unsupervised clustering forest by using Xgboost ? 

Thx

**Comment by lizequn** [comment]
I reposted this on google forum but  on one check, so I reopened this issue here.
